lib/containers/k8s.pm

```diff
@@ -0,0 +1,60 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Basic functionality for testing k3s
+
+package containers::k8s;
+
+use base Exporter;
+use Exporter;
+use strict;
+use warnings;
+use testapi;
+use utils qw(zypper_call script_retry);
+use version_utils qw(is_sle);
+use registration qw(add_suseconnect_product get_addon_fullname);
+
+our @EXPORT = qw(install_k3s uninstall_k3s install_rke2 install_kubectl install_helm test_kubectl);
+
+sub install_k3s {
+    my $k3s_dowload_url = get_required_var("K3S_DOWNLOAD_URL");
+    assert_script_run("curl -LO $k3s_dowload_url");
+    assert_script_run("install -c -m 744 ./k3s /usr/bin/k3s");
```

Comment:
Since this is a custom binary, I would recommend to install it to `/usr/local/bin` instead of `/usr/bin`. Both are typically in `$PATH`, so this should be the only place where modifications are necessary.